Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -195,13 +191,14 @@ void UPSHUDComponent::UpdateProgressionWidgetForPlayer()
 //Is called when local player character is ready to guarantee that they player controller is initialized for the Widget SubSystem
 void UPSHUDComponent::OnLocalCharacterReady_Implementation(APlayerCharacter* Character, int32 CharacterID)
 {
-	// Create widgets now as fast as possible, later we will register them in Widgets Subsystem
-	UWidgetsSubsystem& WidgetsSubsystem = UWidgetsSubsystem::Get(Character);
-	ProgressionMenuWidgetInternal = WidgetsSubsystem.CreateManageableWidgetChecked<UPSMenuWidget>(UPSDataAsset::Get().GetProgressionMenuWidget());
-	ProgressionMenuOverlayWidgetInternal = WidgetsSubsystem.CreateManageableWidgetChecked<UPSOverlayWidget>(UPSDataAsset::Get().GetProgressionOverlayWidget());
+	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	if (PlayerCharacter && !PlayerCharacter->IsLocallyControlled())
+	{
+		return;
+	}
 
-	// Update the progression widget based on current player state
-	UpdateProgressionWidgetForPlayer();
+	UPSWorldSubsystem::Get().OnWorldSubSystemInitialize();
+	UPSWorldSubsystem::Get().OnInitialize.AddDynamic(this, &ThisClass::OnInitialized);
```

Comment:
Order of calls here look suspecious:
1. You call OnWorldSubSystemInitialize
2. Inside, it broadcast OnInitiailize delegate
3. Only now you bind to OnInitiailize which was already broadcasted at step #2

I suspect order here should be changed: first bind to delegate, then broadcast it

Before:
```
	UPSWorldSubsystem::Get().OnWorldSubSystemInitialize();
	UPSWorldSubsystem::Get().OnInitialize.AddDynamic(this, &ThisClass::OnInitialized);
```

After:
```
	UPSWorldSubsystem::Get().OnInitialize.AddDynamic(this, &ThisClass::OnInitialized);
	UPSWorldSubsystem::Get().OnWorldSubSystemInitialize();
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/67874141fbb83e944ed2b656e4a835052e280812